### DISCUSSION TITLE
manual update argoCD between v2.21 to v2.4

### DISCUSSION THREAD

**Discussion by jcsferraz** [argoproj/argo-cd#9974] [Q&A] [answered]
Hello everyone !!

I have questions about the update process at ArgoCD v2.2 to v2.4

follow my currently version:

{
    "Version": "v2.2.1+122ecef",
    "BuildDate": "2021-12-17T01:31:40Z",
    "GitCommit": "122ecefc3abfe8b691a08d9f3cecf9a170cc8c37",
    "GitTreeState": "clean",
    "GoVersion": "go1.16.11",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v4.2.0 2021-06-30T22:49:26Z",
    "HelmVersion": "v3.7.1+g1d11fcb",
    "KubectlVersion": "v0.22.2",
    "JsonnetVersion": "v0.17.0"
}

I read about the process at https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.1-2.2/, but for me it very summarized.

01 - Question

is the argocd update linear?
That is, do I need to upgrade from v2.2 to v2.3 until reaching v2.4?

02 - Question

There are docs, or links to help me step by step to update ?

Thanks a lot!!!

**Comment by crenshaw-dev** [comment, marked as answer]
The update is not linear. You can upgrade directly from 2.1 to 2.4.

However, important changes were made at each step. So read all of these guides and catalog a list of applicable changes / tests you need to make before upgrading in production:

* https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.1-2.2/
* https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.2-2.3/
* https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.3-2.4/

There were some particularly important changes in 2.4, so we wrote a blog post to reiterate/clarify what is in the upgrade notes: https://blog.argoproj.io/breaking-changes-in-argo-cd-2-4-29e3c2ac30c9

> There are docs, or links to help me step by step to update ?

There aren't really step-by-step update docs, because it depends heavily on your deployment. If I install Argo CD using the kustomize docs and kubectl, the upgrade is as simple as `kubectl apply`. If I use Helm + Helm CLI, I use `helm install`, and if I use Helm + Argo CD to bootstrap, I modify an Application and sync it. But it always boils down to: the manifests need to be updated.
